src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,169 @@
+#include "depthai/depthai_wrapper.hpp"
+
+namespace DepthAI
+{
+
+DepthAI::DepthAI(std::string usb_device, std::string config_file, bool usb2_mode) :
+      Device(usb_device, usb2_mode)
+{
+
+    std::ifstream file(config_file);
+    std::ostringstream file_stream;
+    file_stream << file.rdbuf();
+    std::string config_str = file_stream.str();
+
+    pipeline_ = this->create_pipeline(config_str);
+    config_json_ = nlohmann::json::parse(config_str);
+
+    this->set_resolution();
+    this->create_frame_holders();
+
+    width_map_[720] = 1280;
+    width_map_[800] = 1280;
+    width_map_[400] = 640;
+}
+
+void DepthAI::set_resolution()
+{
+
+    if (config_json_.contains("camera"))
+    {
+        auto& camera_conf_obj = config_json_.at("camera");
+        if (camera_conf_obj.contains("rgb"))
+        {
+            auto& rgb_camera_conf_obj = camera_conf_obj.at("rgb");
+            rgb_height_ = rgb_camera_conf_obj.at("resolution_h").get<int32_t>();
+            // _rgb_width = width_map_[_rgb_height];
+        }
+        else
+        {
+            rgb_height_ = 3040;
+            rgb_width_ = 4056;
```

Comment:
Ah all these numbers should really be `static const`s or `constexpr` and not hardcoded in the code. This is for all numbers in this function.

Reply:
how will I modify if it is const

Reply:
What I meant was that rvalues should be constants. For example 3040 should be `static const int  DefaultFrameWidth` or something like that.